src/SocketHttpAdapter.php

```diff
@@ -40,11 +40,19 @@ protected function doSendInternalRequest(InternalRequestInterface $internalReque
     {
         list($protocol, $host, $port, $path) = $this->parseUrl($url = (string) $internalRequest->getUrl());
 
+        $contextOptions = array(
+            'allow_self_signed' => !$this->getConfiguration()->getSSLVerifyPeer(),
+        );
+
+        $context = stream_context_create(array(), $contextOptions);
```

Comment:
If react is not able to do it, please open an issue if there is none and document it as not usable with this adapter.